File: src/js/src/modules/HtmlEntityModule.js
function HtmlEntityModule() {
	HtmlEntityModule.$parent.call(this);
	
	this.value = "";
	
	this.render = function(writer) {
		if (!this.value) {
			throw new ModuleException("The \"value\" property must be set to a valid HTML entity");
		}
		
		if (!/^[a-zA-Z0-9#]+$/.test(this.value)) {
			throw new ModuleException("\"" + this.value + "\" is not a valid HTML entity");
		}
		
		writer.write("&" + this.value + ";");
	};
}

extend(Module, HtmlEntityModule);

File: src/dotnet/Src/ButterflyNet.Parser/Modules/HtmlEntityModule.cs
using System.IO;
using System.Text.RegularExpressions;

namespace ButterflyNet.Parser.Modules {
	public class HtmlEntityModule : IButterflyModule {
		private static readonly Regex entityValidator = new Regex("^[a-zA-Z0-9#]+$");
		public string Value { get; set; }

		public void Render(TextWriter writer) {
			if (string.IsNullOrEmpty(Value)) {
				throw new ModuleException("The \"value\" property must be set to a valid HTML entity");
			}

			if (!entityValidator.IsMatch(Value)) {
				throw new ModuleException(string.Format("\"{0}\" is not a valid HTML entity", Value));
			}

			writer.Write(string.Format("&{0};", Value));
		}
	}
}